src/pages/api/dsr/access.ts

```diff
@@ -0,0 +1,147 @@
+            },
+        });
+    }
+
+    const zip = new JSZip();
+
+    for (const proceeding of proceedings) {
+        const baseDir = zip.folder(proceeding.token);
+        if (!baseDir) throw new Error('This should never happen.');
+
+        if (proceeding.userNetworkActivityRaw || proceeding.uploads.length > 0) {
+            const uploadsDir = baseDir.folder('uploads');
+            if (!uploadsDir) throw new Error('This should never happen.');
+
+            for (const upload of proceeding.uploads) uploadsDir.file(upload.filename, upload.file);
+            if (proceeding.userNetworkActivityRaw)
+                uploadsDir.file('user-network-activity', proceeding.userNetworkActivityRaw);
+        }
+
+        if (proceeding.initialAnalysis?.har) baseDir.file('initial-analysis.har', proceeding.initialAnalysis.har);
+        if (proceeding.secondAnalysis?.har) baseDir.file('second-analysis.har', proceeding.secondAnalysis.har);
+
+        const proceedingPdf = await compileTypst({
+            mainContent: typstFiles.proceeding,
+            additionalFiles: {
+                '/style.typ': typstFiles.style,
+                '/proceeding.json': JSON.stringify(proceeding),
+                '/dpas.json': JSON.stringify(dpas),
+            },
+        });
+        baseDir.file('proceeding.pdf', proceedingPdf);
+
+        if (dataPortabilityRequest) {
+            const p = proceeding as PartialDeep<typeof proceeding>;
+
+            delete p.initialAnalysis?.har;
+            delete p.secondAnalysis?.har;
+            for (const upload of p.uploads || []) delete (upload as Partial<typeof upload>).file;
+
+            baseDir.file(
+                'proceeding.json',
+                JSON.stringify(
+                    p,
+                    (k, v) => {
+                        if (v instanceof Uint8Array) return Buffer.from(v).toString('base64');
+
+                        return v;
+                    },
+                    4,
+                ),
+            );
+        }
+    }
+
+    const letterPdf = await compileTypst({
+        mainContent: typstFiles.accessData,
+        additionalFiles: {
+            '/book-info.svg': bookInfoIcon,
+            '/style.typ': typstFiles.style,
+            '/access.json': JSON.stringify({
+                requestDate: formatDate(requestDate, { language: currentLocale }),
+                reference,
+                responseDate: formatDate(new Date(), { language: currentLocale }),
+                proceedingTokens,
+                dataPortabilityRequest,
+                allTokensFound: proceedings.length === proceedingTokens.length,
+            }),
+        },
+    });
+    zip.file('README.pdf', letterPdf);
+
+    return new Response(await zip.generateAsync({ type: 'nodebuffer' }), {
+        headers: {
+            'Content-Type': 'application/zip',
+            'Content-Disposition': `attachment; filename="${reference}-tweasel-access-request.zip"`,
+        },
+    });
```

Comment:
I mean, it is great that we supply all of this as PDF also, but I really think we should also show this just on the website. This is much more accessible and easier to use e.g. on a mobile device. Not everyone needs a `.zip` of all of their data each time they request access.